src/main/app/controller/GetController.ts

```diff
@@ -114,4 +115,23 @@ export class GetController {
   protected getEventName(req: AppRequest): string {
     return CITIZEN_UPDATE;
   }
+
+  /* eslint-disable @typescript-eslint/ban-types, @typescript-eslint/no-explicit-any */
+  protected redirect(req: AppRequest<any>, res: Response, nextUrl?: string): void {
+    let target;
+    if (req.session.errors?.length) {
+      //redirects to same page in case of validation errors
+      target = req.url;
+    } else {
+      //redirects to input nextUrl if present otherwise calls getNextStepUrl to get the next step url
+      target = nextUrl || getNextStepUrl(req, req.session.userCase);
+    }
+
+    req.session.save(err => {
+      if (err) {
+        throw err;
+      }
+      res.redirect(target);
+    });
+  }
```

Comment:
this is not required
you can use the `saveSessionAndRedirect` method with callback option